### ISSUE TITLE
💡 New  rule: Explicit return type for updater 

### ISSUE THREAD

**Issue by timdeschryver** [timdeschryver/eslint-plugin-ngrx#152]
## Information about the new rule

Similar to [on-function-explicit-return-type](https://github.com/timdeschryver/eslint-plugin-ngrx/blob/main/docs/rules/on-function-explicit-return-type.md), but then for a component-store updater. More info at https://github.com/ngrx/platform/issues/2851.

## A snippet of incorrect code

```ts
updateCount = this.store.updater<number>((state, count) => ({
  ...state,
//  👇 this does not throw.
  count
}));
```

## A snippet of correct code

```ts
updateCount = this.store.updater<number>((state, count): AppComponentState => ({
  ...state,
//  👇 this does throw.
  count
}));
```

## I would be willing to submit a PR for the docs :heart:

[ ] Yes (Assistance is provided if you need help submitting a pull request)
[ ] No

**Comment by timdeschryver** [comment]
Closed by #153
